### PR TITLE
docs: add openapi schemes and docs using huma

### PR DESCRIPTION
**Pull request by mzeevi**

### CODE REVIEW

File: src/routes/v1/doc/operation/terminal.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+				Name:     secWebSocketVersionHeaderKey,
+				In:       headerKey,
+				Required: true,
+				Schema:   huma.SchemaFromType(registry, reflect.TypeOf(huma.TypeString)),
+				Example:  secWebSocketVersionValue,
+			},
+		},
+		RequestBody: &huma.RequestBody{
+			Content: map[string]*huma.MediaType{
+				applicationJSONKey: {
```
Comment (dvirgilad) [inline src/routes/v1/doc/operation/terminal.go:126]: Are we sure this is application/json?
Does this work?

File: src/routes/v1/setup.go [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
 // SetupRoutes initializes the API routes for version 1.
 func SetupRoutes(engine *gin.Engine, tokenProvider auth.TokenProvider, scheme *runtime.Scheme) {
 	engine.Use(middleware.ErrorHandlingMiddleware())
 	v1 := engine.Group("/v1")
 
+	api, r := doc.SetupAPIRegistry(engine)
+
 	engine.GET("/healthz", func(c *gin.Context) {
-		c.JSON(http.StatusOK, gin.H{"message": "ok"})
+		c.JSON(http.StatusOK, types.HealthzResponse{Message: "ok"})
```
Comment (dvirgilad) [inline src/routes/v1/setup.go:24]: Cant we make this messageResponse? seems redundant to have the same type twice

File: src/types/healthz.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package types
+
+type HealthzResponse struct {
```
Comment (dvirgilad) [inline src/types/healthz.go:3]: This type is Exactly the same as MessageResponse. Maybe we can just do 
```go
type HealthzResponse MessageResponse
```

File: src/types/capp.go [hunk 4]
```diff
@@ -64,7 +64,7 @@ type CappUri struct {
 	CappName      string `uri:"cappName" binding:"required"`
 }
 
-type CappError struct {
+type MessageResponse struct {
```
Comment (dvirgilad) [inline src/types/capp.go:67]: Seeing as this is more general than the rest of the types here. It should be moved to common.go

**Review by dvirgilad — CHANGES_REQUESTED** [review]
Looks good,
Left some minor stuff

File: src/routes/v1/doc/operation/terminal.go (hunk 1 above)
Comment (mzeevi) [reply src/routes/v1/doc/operation/terminal.go:126]: So none of the websockets stuff (logs/terminal) actually work from the `/docs` endpoint, this serves more as an example to use in clients that can actually do websockets (i.e. Postman)

**Review by dvirgilad — APPROVED** [review]
/lgtm

**Pull request merged**